build/utils/dotnet/solution/solution.go

```diff
@@ -174,14 +174,27 @@ func (solution *solution) loadProjects(slnProjects []project.Project, log utils.
 	return nil
 }
 
-func (solution *solution) parseProjectsFromSolutionFile(slnProjects []string, log utils.Log) ([]project.Project, error) {
+func (solution *solution) parseProjectsFromSolutionFile(slnProjects []string, excludePattern string, log utils.Log) ([]project.Project, error) {
 	var projects []project.Project
 	for _, projectLine := range slnProjects {
 		projectName, projFilePath, err := parseProjectLine(projectLine, solution.path)
 		if err != nil {
 			log.Error(err)
 			continue
 		}
+		// Exclude projects by pattern.
+		if len(excludePattern) > 0 {
+			log.Debug(fmt.Sprintf("Testing to exclude projects by pattern: %s", excludePattern))
+			exclude, err := regexp.MatchString(excludePattern, projFilePath)
+			if err != nil {
+				log.Error(err)
+				continue
```

Comment:
I think an error is OK in this case, we log it just like we log when we can't parse at the above line.
There could be an error from the pattern or a lot of other reasons...
We got an error for this project so we report it and skip to the next, we don't want to fail all the build but still an error occur and it is not the full one, the customer may be at fault since we rely on input given to us.